Server/mods/deathmatch/logic/packets/CPedSyncPacket.cpp

```diff
@@ -13,22 +13,38 @@
 
 CPedSyncPacket::~CPedSyncPacket()
 {
-    vector<SyncData*>::const_iterator iter = m_Syncs.begin();
-    for (; iter != m_Syncs.end(); ++iter)
+    for (auto iter = m_Syncs.begin(); iter != m_Syncs.end(); ++iter)
     {
         delete *iter;
     }
     m_Syncs.clear();
 }
 
+CPedSyncPacket::CPedSyncPacket(SyncData* pReadData)
+{
+    // Create a new struct and copy the data
+    SyncData* pWriteData = new SyncData;
+    pWriteData->Model = pReadData->Model;
+    pWriteData->ucFlags = pReadData->ucFlags;
+    pWriteData->ucSyncTimeContext = pReadData->ucSyncTimeContext;
+    pWriteData->vecPosition = pReadData->vecPosition;
+    pWriteData->fRotation = pReadData->fRotation;
+    pWriteData->vecVelocity = pReadData->vecVelocity;
+    pWriteData->fHealth = pReadData->fHealth;
+    pWriteData->fArmor = pReadData->fArmor;
+    pWriteData->bOnFire = pReadData->bOnFire;
+    pWriteData->bIsInWater = pReadData->bIsInWater;
```

Comment:
Why not just use `SyncData`'s copy constructor?
Also, I know this is more like a refactor, but I'd suggest using a `std::vector<SyncData>` instead of `SyncData*`. I'd that would be faster in the long run.

Reply:
> Why not just use `SyncData`'s copy constructor?

I was tripping

> Also, I know this is more like a refactor, but I'd suggest using a `std::vector<SyncData>` instead of `SyncData*`. I'd that would be faster in the long run.

Yeah ok, I don't know which is fastest tbh. I changed it to reference `std::vector <SyncData>`. (removed the destructor)

Reply:
Well, what you need to consider is that your `vector` will reallocate in power by 2 steps. What you could do to make it optimal is to `reserve` a specific capacity ahead of time, that would make it infinitely faster.

Reply:
Alright, reserved a sensible amount. (This is for m_nearPlayersList, not SyncData)